### DISCUSSION TITLE
Is it possible to ensure that the message will not be lost when the topic has no consumers

### DISCUSSION THREAD

**Discussion by qcjxberin** [dotnet/MQTTnet#1274] [Q&A]
Is it possible to ensure that the message will not be lost when the topic has no consumers

**Comment by biapar** [comment]
Use qos 1 or 2 and retain messages option.

**Comment by chkr1011** [comment]
@qcjxberin You can attach to the event _ApplicationMessageNotConsumedAsync_. It is fired whenever a message was not consumed by any client.
